sqlglot/optimizer/qualify_columns.py

```diff
@@ -240,13 +240,18 @@ def _expand_alias_refs(scope: Scope, resolver: Resolver, expand_only_groupby: bo
     def replace_columns(
         node: t.Optional[exp.Expression], resolve_table: bool = False, literal_index: bool = False
     ) -> None:
-        if not node or (expand_only_groupby and not isinstance(node, exp.Group)):
+        is_group_by = isinstance(node, exp.Group)
+        if not node or (expand_only_groupby and not is_group_by):
             return
 
         for column in walk_in_scope(node, prune=lambda node: node.is_star):
             if not isinstance(column, exp.Column):
                 continue
 
+            # BigQuery should expand aliased expressions only if the alias is used as a standalone name
+            if expand_only_groupby and is_group_by and column.parent is not node:
+                continue
```

Comment:
Is bigquery able to only expand aliases within `GROUP BY`, or other clauses too?

Reply:
I'd consider including this context in the comment in L251.

Reply:
Sg, tried to summarize these 2 concepts in that comment